content/docs/develop/userscripts/about-userscripts.md

```diff
@@ -0,0 +1,142 @@
+```js
+// Example userscript
+export default async function ({ addon, console }) {
+  console.log("Hello, " + await addon.auth.fetchUsername());
+  console.log("How are you today?");
+}
+```
+
+Remember that JavaScript allows functions to be declared inside other functions, for example:
+```js
+export default async function ({ addon, console }) {
+  async function sayHelloToUser() {
+    console.log("Hello, " + await addon.auth.fetchUsername());
+  }
+
+  await sayHelloToUser();
+  console.log("How are you today?");
+}
+```
+
+{{< admonition info >}}
+You can access many `addon.*` API utilities from userscripts. For example, you can get the current username, wait until an element exists on the page, or get a reference to the Scratch VM object.
+
+For more information, check the [API reference](/docs/reference/addon-api/).
+{{< /admonition >}}
+
+
+## Modifying the document HTML
+
+Use [browser DOM APIs](https://developer.mozilla.org/en-US/docs/Web/API/HTML_DOM_API) to customize the HTML of the page.
+
+<!-- TODO : an example -->
+
+## Localizing userscripts
+
+Addon userscripts sometimes need to reference English words or sentences. Make sure not to hardcode them in the code, so that they can be part of the translation process.
+
+{{< admonition error >}}
+```js
+// Don't do this:
+document.querySelector(".sa-find-bar").placeholder = "Find blocks";
+```
+{{< /admonition >}}
+
+To create a translatable string, follow these steps:
+1. Create a file named `addon-id.json` inside the `/addon-l10n/en` folder.
+2. Provide an ID for every string:
+```json
+{
+  "addon-id/find": "Find blocks"
+}
+```
+3. Make sure to import the `msg()` function in your userscript. The first line of your userscript should look like this:
+```js
+export default async function ({ addon, console, msg  }) {
+                                              // ^^^
+```
+4. Use the `msg()` function in your code, instead of a hardcoded string:
+```js
+document.querySelector(".sa-find-bar").placeholder = msg("find");
+```
+
+{{< admonition info >}}
+For more information about localizing userscripts, see [this page](/docs/localization/localizing-addons/).
+{{< /admonition >}}
+
+
+## Technical details
+
+Each userscript file is a JavaScript module that exports a function. Scratch Addons only imports the module if needed, and executes it after the page has fully loaded.
+
+<!-- TODO: explain runAtComplete -->
+
+Userscripts are JavaScript modules, so they always run on ["strict mode"](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode). This also means that userscripts may use [top-level imports](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) to import other JavaScript files.
+
+<!-- TODO: explain execution order isn't guaranteed -->
+
+
+## Debugging userscripts
+
+<!-- TODO: revisit this section -->
+
+**Make sure to refresh Scratch Addons from `chrome://extensions` after making any changes to your addon.**  
```

Comment:
This kind of contradicts with the beginning of the page, but this whole section will probably be redone at some point anyway.

Reply:
Yeah, this is part of the old doc as well.

Reply:
This section has been moved to a new page, and this sentence has been removed.